components/21-atoms/table/table.functions.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+/* global define, module */
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define(factory);
+  }
+  else {
+    if (typeof exports === 'object') {
+      module.exports = factory();
+    }
+    else {
+      root.ResponsiveTable = factory();
+    }
+  }
+}(this || window, function () {
+  return function (element, options) {
+
+    /**
+     * The table caption.
+     */
+    let caption = null;
+
+    /**
+     * Determine if the table should be focusable.
+     */
+    const determineFocusable = () => {
+      var scrollableWidth = element.parentNode.querySelector('table').scrollWidth;
+      var containerWidth = element.parentNode.clientWidth;
+
+      // Check if element is scrollable.
+      if (scrollableWidth <= containerWidth) {
+        // If not remove the tab focus.
+        element.removeAttribute('tabindex');
+      }
+      else {
+        caption.innerText += ' ' + options.scrollableText;
+      }
+    };
+
+    /**
+     * Setup the responsive table.
+     */
+    const setupResponsiveTable = () => {
+      // Set table container attributes.
+      element.setAttribute('tabindex', '0');
+      element.setAttribute('role', 'group');
+      element.setAttribute('aria-labelledby', 'caption');
+
+      // Set caption id.
+      if (!caption.hasAttribute('id')) {
+        caption.setAttribute('id', 'caption');
```

Comment:
you shouldn't set the ID to something so general. If 2 tables exist on the same page, you'll create invalid HTML.

Reply:
👍